src/test/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndexTest.java

```diff
@@ -1,19 +1,29 @@
 package org.broadinstitute.hellbender.utils.bwa;
 
 import org.testng.Assert;
+import org.testng.Reporter;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Random;
+import java.util.*;
+import java.util.stream.Stream;
 
 @Test
 public final class BwaMemIndexTest {
```

Comment:
It won't work.  There is stochasticity in the index creation that prevents direct comparison.
There is a pending PR, which I surely which @droazen would review since it's been stalled for weeks, that embeds a magic word, a length, a CRC, and a bwa version in the index file so that we can make sure everything is good to go.